### PR TITLE
test: add tests for file-upload components

### PR DESCRIPTION
**Pull request by saumyatalwani**
> [!IMPORTANT]
>
> 1. Make sure you have read our [contribution guidelines](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md)
> 1. Ensure there is an associated issue and you have been assigned to it
> 1. Use the correct syntax to link this PR: `Fixes #<issue number>`.

## Summary

* **Tests**
  * Vastly expanded test coverage for the file uploader: inputs (local/link), lists, image/audio/video/PDF previews, attachment/chat integrations, store/hooks, utils, hotkeys, portals, upload/reupload flows, size/type limits and many edge cases.
* **Refactor**
  * Replaced icon components with span-based icon fonts and added test IDs for key actions.
* **Chores**
  * Removed select ESLint suppression entries.

| File | Statements | Branches | Functions | Lines |
|---|---|---|---|---|
| `app/components/base/file-uploader/audio-preview.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/constants.ts` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/file-from-link-or-local/index.tsx` | 95.45% | 91.3% | 100% | 95% |
| `app/components/base/file-uploader/file-image-render.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/file-input.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/file-list-in-log.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/file-type-icon.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/file-uploader-in-attachment/file-item.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/file-uploader-in-attachment/index.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/file-uploader-in-chat-input/file-image-item.tsx` | 100% | 95.83% | 100% | 100% |
| `app/components/base/file-uploader/file-uploader-in-chat-input/file-item.tsx` | 100% | 97.87% | 100% | 100% |
| `app/components/base/file-uploader/file-uploader-in-chat-input/file-list.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/file-uploader-in-chat-input/index.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/hooks.ts` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/pdf-preview.tsx` | 100% | 100% | 100% | 100% |
| `app/components/base/file-uploader/store.tsx` | 100% | 85.71% | 100% | 100% |
| `app/components/base/file-uploader/utils.ts` | 100% | 99% | 100% | 100% |
| `app/components/base/file-uploader/video-preview.tsx` | 100% | 100% | 100% | 100% |

## Screenshots

| Before | After |
|--------|-------|
| ... | ... |

## Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `make lint` and `make type-check` (backend) and `cd web && npx lint-staged` (frontend) to appease the lint gods

Fixes: #32152

### CODE REVIEW

File: web/app/components/base/file-uploader/file-uploader-in-chat-input/file-item.tsx [hunk 1]
```diff
@@ -62,20 +57,21 @@ const FileItem = ({
...
+              data-testid="delete-button"
             >
-              <RiCloseLine className="h-4 w-4 text-components-button-secondary-text" />
+              <span className="i-ri-close-line h-4 w-4 text-components-button-secondary-text" />
             </Button>
           )
         }
         <div
-          className="system-xs-medium mb-1 line-clamp-2 h-8 cursor-pointer break-all text-text-tertiary"
+          className="mb-1 line-clamp-2 h-8 cursor-pointer break-all text-text-tertiary system-xs-medium"
```
[the remark this answers is not in the document]
Comment (saumyatalwani) [reply web/app/components/base/file-uploader/file-uploader-in-chat-input/file-item.tsx:67]: not relevant to this PR

File: web/app/components/base/file-uploader/file-uploader-in-chat-input/file-item.tsx [hunk 2]
```diff
@@ -118,10 +115,7 @@ const FileItem = ({
           }
           {
             uploadError && (
-              <ReplayLine
-                className="h-4 w-4 text-text-tertiary"
-                onClick={() => onReUpload?.(id)}
-              />
+              <span className="i-custom-vender-other-replay-line h-4 w-4 cursor-pointer text-text-tertiary" onClick={() => onReUpload?.(id)} data-testid="replay-icon" role="button" tabIndex={0} />
```
[the remark this answers is not in the document]
Comment (saumyatalwani) [reply web/app/components/base/file-uploader/file-uploader-in-chat-input/file-item.tsx:118]: not relevant to this PR, needs a separate PR

File: web/app/components/base/file-uploader/audio-preview.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import AudioPreview from './audio-preview'
+
+vi.mock('react-hotkeys-hook', () => ({
```
Comment (CodingOnStar) [inline web/app/components/base/file-uploader/audio-preview.spec.tsx]: please avoid mocking third-party libraries.

File: web/app/components/base/file-uploader/pdf-preview.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
+import type { ReactNode } from 'react'
+import { act, render, screen } from '@testing-library/react'
+import PdfPreview from './pdf-preview'
+
+const mockUseHotkeys = vi.fn()
+vi.mock('react-hotkeys-hook', () => ({
```
Comment (CodingOnStar) [inline web/app/components/base/file-uploader/pdf-preview.spec.tsx]: and here

File: web/app/components/base/file-uploader/video-preview.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import { fireEvent, render } from '@testing-library/react'
+import VideoPreview from './video-preview'
+
+vi.mock('react-hotkeys-hook', () => ({
```
Comment (CodingOnStar) [inline web/app/components/base/file-uploader/video-preview.spec.tsx]: and here

**Comment by saumyatalwani** [comment]
@CodingOnStar added adapter for pdf-highlighter as direct react-pdf-highlighter import was failing under Vitest in this repo due to a broken internal pdfjs-dist resolution path.

**Review by CodingOnStar — APPROVED** [review]
